src/terraform.yml

```diff
@@ -39,14 +39,16 @@ repository:
       - include: "#expressions"
 
   functions:
-    begin: (\w+)(\()
+    begin: ([:\-\w]+)(\()
     name: meta.function-call.hcl
     comment: Built-in function calls
     beginCaptures:
       "1":
         patterns:
           - match: \b(abs|abspath|alltrue|anytrue|base64decode|base64encode|base64gzip|base64sha256|base64sha512|basename|bcrypt|can|ceil|chomp|chunklist|cidrhost|cidrnetmask|cidrsubnet|cidrsubnets|coalesce|coalescelist|compact|concat|contains|csvdecode|dirname|distinct|element|endswith|file|filebase64|filebase64sha256|filebase64sha512|fileexists|filemd5|fileset|filesha1|filesha256|filesha512|flatten|floor|format|formatdate|formatlist|indent|index|join|jsondecode|jsonencode|keys|length|log|lookup|lower|matchkeys|max|md5|merge|min|nonsensitive|one|parseint|pathexpand|plantimestamp|pow|range|regex|regexall|replace|reverse|rsadecrypt|sensitive|setintersection|setproduct|setsubtract|setunion|sha1|sha256|sha512|signum|slice|sort|split|startswith|strcontains|strrev|substr|sum|templatefile|textdecodebase64|textencodebase64|timeadd|timecmp|timestamp|title|tobool|tolist|tomap|tonumber|toset|tostring|transpose|trim|trimprefix|trimspace|trimsuffix|try|upper|urlencode|uuid|uuidv5|values|yamldecode|yamlencode|zipmap)\b
             name: support.function.builtin.terraform
+          - match: \bprovider::(?!false|true)\w*::(?!null|false|true)\w*\b
```

Comment:
I'm not sure if replacing everything with just a single `\w` is a good thing here. AFAIK functions can't start with a number or `-` or `_`. That's why there used to be a `[:alpha:]` regex for the first character before. Also `\w` also doesn't contain `-`, which was allowed before. 

Are you sure provider names can't start with `false` or `true`? If not, I think we can remove the negative look ahead.

We can also discuss this over zoom, if you prefer.

Reply:
Changed it to the previous one, but replaced `[[:alpha:]][[:alnum:]_-]*` with `[^\d\W][\w_-]*`
I'm looking into whether `false` or `true` are allowed, will remove them from the regex if they are.

Reply:
Removed `true`, `false`, and `null` as they are allowed as far as Terraform is concerned.